File: src/main/java/com/clinichub/entity/PrescriptionItem.java
package com.clinichub.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "prescription_item")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class PrescriptionItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "prescription_id", nullable = false)
    @JsonIgnore
    private Prescription prescription;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "herb_id", nullable = false)
    private Herb herb;

    @Column(name = "dose_g", nullable = false)
    private Integer doseG;

    @Column(length = 100)
    private String note;
}

File: flutter_app/lib/models/prescription_item.dart
import 'herb.dart';

class PrescriptionItem {
  final int id;
  final Herb herb;
  final int doseG;
  final String? note;

  PrescriptionItem({
    required this.id,
    required this.herb,
    required this.doseG,
    this.note,
  });

  factory PrescriptionItem.fromJson(Map<String, dynamic> json) => PrescriptionItem(
    id: json['id'],
    herb: Herb.fromJson(json['herb']),
    doseG: json['doseG'],
    note: json['note'],
  );
}
